x-pack/test/fleet_api_integration/apis/agents/reassign.ts

```diff
@@ -209,17 +209,37 @@ export default function (providerContext: FtrProviderContext) {
           })
           .expect(200);
 
-        expect(unenrolledBody).to.eql({
-          agent1: { success: true },
-          agent2: { success: true },
-          agent3: { success: true },
-          agent4: { success: true },
-        });
+        const actionId = body.actionId;
 
-        const { body } = await supertest.get(`/api/fleet/agents`).set('kbn-xsrf', 'xxx');
-        expect(body.total).to.eql(4);
-        body.items.forEach((agent: any) => {
-          expect(agent.policy_id).to.eql('policy2');
+        const verifyActionResult = async () => {
+          const { body: result } = await supertest.get(`/api/fleet/agents`).set('kbn-xsrf', 'xxx');
+          expect(result.total).to.eql(4);
+          result.items.forEach((agent: any) => {
+            expect(agent.policy_id).to.eql('policy2');
+          });
+        };
+
+        await new Promise((resolve, reject) => {
+          let attempts = 0;
+          const intervalId = setInterval(async () => {
```

Comment:
For testing the async execution of bulk actions, I had to wait for a Promise, otherwise the FTR shutdown starts and the action doesn't finish.
Added an interval of 1s for 3 times to check whether the action has completed.